### PR TITLE
Rule ID change to GUID

### PR DESCRIPTION
**Pull request by noodlemoodle**
migrate rules to have long ids generated by guid instead of user-defined strs

nano side https://github.com/nanoPayinc/NANOPAY/pull/9354

### CODE REVIEW

File: src/foam/nanos/ruler/Rule.js [hunk 1]
```diff
@@ -380,8 +386,8 @@
       name: 'javaExtras',
       buildJavaClass: function(cls) {
         cls.extras.push(`
-        public static Rule findById(Collection<Rule> listRule, String passedId) {
-          return listRule.stream().filter(rule -> passedId.equals(rule.getId())).findFirst().orElse(null);
+        public static Rule findById(Collection<Rule> listRule, Long id) {
```
Comment (kgrgreer) [inline src/foam/nanos/ruler/Rule.js]: Why Long instead of long?

File: src/foam/nanos/ruler/RuleEngine.java [hunk 2]
```diff
@@ -150,7 +150,7 @@ public void putResult(Object result) {
     results_.put(currentRule_.getId(), result);
   }
 
-  public Object getResult(String ruleId) {
+  public Object getResult(Long ruleId) {
```
Comment (kgrgreer) [inline src/foam/nanos/ruler/RuleEngine.java]: Long -> long ?

File: src/foam/nanos/ruler/Rule.js (hunk 1 above)
Comment (kgrgreer) [reply src/foam/nanos/ruler/Rule.js]: Where is this called from? Is it performance sensitive? If so, could you use a Map instead?

Comment (noodlemoodle) [reply src/foam/nanos/ruler/Rule.js]: i think this is called from a listener on ruledao in the rulerdao decorator to add to or remove from the rules list of the target dao whenever the ruledao changes

File: src/foam/nanos/ruler/RuleEngine.java (hunk 2 above)
Comment (noodlemoodle) [reply src/foam/nanos/ruler/RuleEngine.java]: sorry, typo

**Comment by kgrgreer** [comment]
LGTM

**Review by jlhughes — CHANGES_REQUESTED** [review]
Ok, miscommunication here. 
Id should be class 'String' and set by GUID. 
We changed then from 'long' to String, as it was impossible to add them from the repository without conflict.

File: src/foam/nanos/ruler/Rule.js [hunk 3]
```diff
@@ -63,7 +65,11 @@
     {
       class: 'String',
       name: 'id',
-      updateVisibility: 'RO',
+      hidden: true
```
Comment (jlhughes) [inline src/foam/nanos/ruler/Rule.js]:
```suggestion
      visibility: 'HIDDEN'
```

**Review by jlhughes — APPROVED** [review]

**Pull request merged**